### PR TITLE
Fix for #3243, #3244 Prevent data loss when client provides pipes to exec

### LINKED ISSUE #3244: ConfigTest.honorClientAuthenticatorCommands fails on Windows
The test fails with parsing error 

```
[main] ERROR io.fabric8.kubernetes.client.Config - Failed to parse the kubeconfig.
expected '<document start>', but found '<scalar>'
 in 'reader', line 1, column 5:
    '.' is not recognized as an internal ...
        ^
```
Because the output from `cmd /c ./token-generator-win` is `'.' is not recognized as an internal or external command,
operable program or batch file.`

The config file needs to use platform's path separator, which is backslash. 
Issue will be fixed alongside #3243

### PR DESCRIPTION
**Pull request by pdudits**
## Description

Fix #3244 #3243 

This fixes data loss when using `PipedInputStream` to pump output from exec command. Since this user-provided resource was auto-closed on websocket connection close, client was unable to read contents of the buffer after last chunk was received. (#3243)

Handling of `PipedOutputStream` provided for `stdin` was changed in same fashion, although the impact there is not that big due to how the pair is implemented.

I've run the unit tests in my Windows environment and fixed any issues with tests I encountered there (none of them were related to pipes, rather OS quirks). (#3244)

## Type of change

 - [x] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [x] I Added [CHANGELOG](../CHANGELOG.md) entry regarding this change
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [x] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Review by shawkins — CHANGES_REQUESTED** [review]
LGTM, but can the imports be left in their original position?  There's another pr open that this will introduce conflicts with.

**Comment by pdudits** [comment]
I didn't notice I had that activated in the IDE, sorry.

Import sections have been reset.

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ExecWebSocketListener.java [hunk 1]
```diff
@@ -349,7 +349,6 @@ private static InputStream inputStreamOrPipe(InputStream stream, PipedOutputStre
             return stream;
         } else if (out != null) {
             PipedInputStream pis = bufferSize == null ? new PipedInputStream() : new PipedInputStream(bufferSize.intValue());
-            toClose.add(out);
```
Comment (iocanel) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ExecWebSocketListener.java:352]: +1: You should never ever be your responsibility to close a stream that was handed out to you.

**Review by iocanel — APPROVED** [review]
LGTM, thanks a lot!

**Review by rohanKanojia — APPROVED** [review]

**Review by shawkins — APPROVED** [review]
LGTM

**Review by manusa — APPROVED** [review]
LGTM, thx!

**Pull request merged**